src/__tests__/lib/containers/widgets/Range.test.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react'
+import { mount, ReactWrapper } from 'enzyme'
+import { Range, RangeProps } from '../../../../lib/containers/widgets/Range'
+
+const mockCallback = jest.fn()
+
+function createTestProps(overrides?: RangeProps): RangeProps {
+  return {
+    type: 'number',
+    initialValues: { min: '2.3', max: '5' },
+    className: 'rangeClass',
+    onChange: mockCallback,
+    ...overrides,
+  }
+}
+let wrapper: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>
+let props: RangeProps
+
+function init(overrides?: RangeProps) {
+  props = createTestProps(overrides)
+  wrapper = mount(<Range {...props} />)
+}
+
+beforeEach(() => init())
+
+function updateValue( wrapper: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>, min: number | string, max: number | string) {
+  wrapper
+  .find('input')
+  .at(0)
+  .simulate('change', { target: { value: min } })
+wrapper
+  .find('input')
+  .at(1)
+  .simulate('change', { target: { value: max } })
+wrapper.find('button').simulate('click')
+}
+
+describe('number range ', () => {
+  it('should render with correct properties', () => {
+    expect(wrapper).toBeDefined()
+    expect(
+      wrapper
+        .find('input')
+        .at(0)
+        .props().value,
+    ).toBe(props.initialValues.min)
+    expect(
+      wrapper
+        .find('input')
+        .at(1)
+        .props().value,
+    ).toBe(props.initialValues.max)
+    expect(
+      wrapper
+        .find('div')
+        .at(0)
+        .hasClass('rangeClass'),
+    ).toBe(true)
+  })
+
+  it('should call callbackFn correctly', async () => {
+    updateValue(wrapper, 2, 4.9)
+    expect(mockCallback).toHaveBeenCalledWith({ min: 2, max: 4.9 })
+  })
+  it('should not trigger callback for min > max and should show error', async () => {
+    updateValue(wrapper, 4.9, 2)
+    expect(mockCallback).not.toHaveBeenCalledWith()
```

Comment:
expect(mockCallback).not.toHaveBeenCalledWith() -> expect(mockCallback).not.toHaveBeenCalled()